### ISSUE TITLE
Changing base href in web assembly blazor has undesired results

### ISSUE THREAD

**Issue by kostasxagoraris** [dotnet/aspnetcore#24235]
I have the same exact issue with 
https://github.com/dotnet/aspnetcore/issues/19928

The thing is that on that issue it is mentioned that it has been resolved on one of the previews but i can still see it there and I cant do much about it.
I can explicitly change the IssuerUri like this

services.AddIdentityServer (options=> { options.IssuerUri = "http://localhost/panasoftweb"; }) 

at startup.cs ConfigureServices

and that will work but this is a very temporary solution.
Because this application is going to be installed on 400 clients on their local iis and i cant possibly know the configuration or even change the appsettings on all those machines or manage every change my clients may make on their iis -web app naming.

### Further technical details
- ASP.NET Core 3.1
- The IDE (VS 2019 16.6.5)

dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.400-preview-015151
 Commit:    0755a9e324

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.400-preview-015151\

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  1.1.13 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.604 [C:\Program Files\dotnet\sdk]
  2.1.801 [C:\Program Files\dotnet\sdk]
  2.1.802 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.400-preview-015151 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by mkArtakMSFT** [comment]
Thank you for filing this issue. Please provide more details about what is the issue you're facing. What you're trying to do and what doesn't work. Also, in order for us to investigate this, please provide a minimalistic repro project (ideally a GitHub repo) that illustrates the problem.

**Comment by kostasxagoraris** [comment]
I don't think I can provide a code sample that will illustrate the problem because the problem is partially caused by the deployment location.
If you follow the exact same instructions as on #19928 and not deploy on http://{localhost}/ but instead http://{localhost}/{application} you will see that every ajax call back to the server fails with 401 error code because for some reason the issuer is considered to be  invalid (untrusted/alien/unknown). 

long story short: Can't deploy to web application (http://{localhost}/{application})  on iis but only on the main web app (http://{localhost}/)

![401](https://user-images.githubusercontent.com/63858034/88552821-333d9100-d02d-11ea-942a-594e77d78131.png)
